### PR TITLE
Raise error on unrecognized flags #205

### PR DESCRIPTION
**Pull request by allieychen**
Raise an error and fail the pipeline if there are unrecognized flags.

A previous fix ([PR 302](https://github.com/googlegenomics/gcp-variant-transforms/pull/302)) cannot handle the case when the args have a `dest` specified (e.g., `--worker_machine_type` has a dest as `mathine_type`, in which case `worker_mathine_type` is not recognized).

Issue:  [#205](https://github.com/googlegenomics/gcp-variant-transforms/issues/205)
Tested: unit tests & integration tests

### CODE REVIEW

File: gcp_variant_transforms/vcf_to_bq_common.py [hunk 1]
```diff
@@ -123,16 +123,10 @@ def write_headers(merged_header, file_path):
...
-  """Returns a list of flag names that starts with `COMMAND_LINE_ARG_PREFIX`."""
-  return [arg.lstrip(_COMMAND_LINE_ARG_PREFIX) for arg in pipeline_args
-          if arg.startswith(_COMMAND_LINE_ARG_PREFIX)]
+  parser = argparse.ArgumentParser()
+  for cls in pipeline_options.PipelineOptions.__subclasses__():
+    if '_add_argparse_args' in cls.__dict__:
+      cls._add_argparse_args(parser)
+  _, unknown = parser.parse_known_args(pipeline_args)
+  if unknown:
+    raise ValueError('The flag {} is unrecognized.'.format(unknown))
```
Comment (arostamianfar) [inline gcp_variant_transforms/vcf_to_bq_common.py]: nit: s/flag/flag(s) as there can be multiple ones...I recommend rephrasing as `Unrecognized flag(s): {}`

File: gcp_variant_transforms/vcf_to_bq_common.py [hunk 2]
```diff
@@ -123,16 +123,10 @@ def write_headers(merged_header, file_path):
...
-    if not any(option.startswith(flag) for option in options):
-      raise ValueError('The flag {} is unrecognized.'.format(flag))
-
-
-def _get_flag_names(pipeline_args):
-  # type: (List[str]) -> List[str]
-  """Returns a list of flag names that starts with `COMMAND_LINE_ARG_PREFIX`."""
-  return [arg.lstrip(_COMMAND_LINE_ARG_PREFIX) for arg in pipeline_args
-          if arg.startswith(_COMMAND_LINE_ARG_PREFIX)]
+  parser = argparse.ArgumentParser()
```
Comment (arostamianfar) [inline gcp_variant_transforms/vcf_to_bq_common.py]: Please also remove the constant `_COMMAND_LINE_ARG_PREFIX`

File: gcp_variant_transforms/vcf_to_bq_common.py (hunk 1 above)
Comment (allieychen) [reply gcp_variant_transforms/vcf_to_bq_common.py]: Thanks! That's better.

File: gcp_variant_transforms/vcf_to_bq_common.py (hunk 2 above)
Comment (allieychen) [reply gcp_variant_transforms/vcf_to_bq_common.py]: Right! Done.

**Pull request merged**